### DISCUSSION TITLE
Ability for user to login without OTP if they have not setup OTP device yet.

### DISCUSSION THREAD

**Discussion by mohsiniscoding** [django-otp/django-otp#149] [Q&A] [answered]
Thanks for the awesome project, I'm trying to add 2FA verification on django admin, as soon as I use OTPAdminSite class, it locked me out asking for OPT to login, is there a way to allow logic if the user hasn't setup OTP device yet?

**Comment by psagers** [comment, marked as answer]
This is probably what you're looking for:

https://django-otp-official.readthedocs.io/en/stable/overview.html#addstatictoken

Also noted here:

https://django-otp-official.readthedocs.io/en/stable/auth.html#the-admin-site

**Comment by mohsiniscoding** [comment]
@psagers Got it thanks, another question, is there a way to not require OTP if user was already logged-in before? (remember me next time - do not ask for otp).

**Reply by psagers** [reply]
See this footnote:

https://django-otp-official.readthedocs.io/en/stable/overview.html#agents

**Comment by mohsiniscoding** [comment]
Good enough, last: what the best way  (least effort) to bind one OTP device to all users (i know this is unsafe but this is very specific use case to our retail app).

**Reply by psagers** [reply]
Each device belongs to one user. In the case of `TOTPDevice`, I suppose you could define a single key somewhere and use it to spawn an identical device for each user. It mostly defeats the purpose of two-factor authentication, but I suppose it might be useful for some kind of malicious compliance. Or you might just write your own device class, which is pretty easy. It's definitely off-label, so you're pretty much on your own.
